azkaban-common/src/main/java/azkaban/utils/EmailMessage.java

```diff
@@ -234,24 +234,35 @@ public void sendEmail() throws MessagingException {
 
     final SMTPTransport t = (SMTPTransport) session.getTransport(protocol);
 
-    try {
-      connectToSMTPServer(t);
-    } catch (final MessagingException ste) {
-      if (ste.getCause() instanceof SocketTimeoutException) {
-        try {
-          // retry on SocketTimeoutException
-          connectToSMTPServer(t);
-          this.logger.info("Email retry on SocketTimeoutException succeeded");
-        } catch (final MessagingException me) {
-          this.logger.error("Email retry on SocketTimeoutException failed", me);
-          throw me;
-        }
-      } else {
-        this.logger.error("Encountered issue while connecting to email server", ste);
-        throw ste;
+    int attempt;
+    for (attempt = 0; attempt < MAX_EMAIL_RETRY_COUNT; attempt++) {
```

Comment:
Breaking this into two helper methods (one for connecting to SMTP server and one for sending the message) might make this a bit more readable.